TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -1011,31 +1024,52 @@ EventBase MapImportedEvent(EntityVersionInfo<dfeta_qualification> snapshot)
                 CreatedUtc = snapshot.Timestamp,
                 RaisedBy = Events.Models.RaisedByUserInfo.FromDqtUser(snapshot.UserId, snapshot.UserName),
                 PersonId = snapshot.Entity.dfeta_PersonId.Id,
-                MandatoryQualification = GetEventMandatoryQualification(snapshot.Entity),
-                OldMandatoryQualification = GetEventMandatoryQualification(previous.Entity),
+                MandatoryQualification = GetEventMandatoryQualification(snapshot.Entity, applyMigrationMappings: false),
+                OldMandatoryQualification = GetEventMandatoryQualification(previous.Entity, applyMigrationMappings: false),
                 ChangeReason = null,
                 ChangeReasonDetail = null,
                 EvidenceFile = null,
                 Changes = changes
             };
         }
 
-        Events.Models.MandatoryQualification GetEventMandatoryQualification(dfeta_qualification snapshot)
+        EventBase MapMigratedEvent(EntityVersionInfo<dfeta_qualification> snapshot)
+        {
+            var eventMandatoryQualification = GetEventMandatoryQualification(snapshot.Entity, applyMigrationMappings: true);
+
+            var changes = MandatoryQualificationMigratedEventChanges.None |
+                (eventMandatoryQualification.Provider?.Name != eventMandatoryQualification.Provider?.DqtMqEstablishmentName ? MandatoryQualificationMigratedEventChanges.Provider : 0);
```

Comment:
So anything with the exact same name will not be seen as a change then?

Reply:
Yes, that’s the idea.